src/index.ts

```diff
@@ -107,6 +106,22 @@ function swc(options: PluginOptions = {}): RollupPlugin {
           ? {}
           : getOptions(this, dirname(id), options.tsconfig);
 
+      let enableExperimentalDecorators = false;
+      if (isTypeScript) {
+        try {
+          const packageJsonPath = require.resolve('typescript/package.json', { paths: [process.cwd()] });
+          const { version } = JSON.parse(fs.readFileSync(packageJsonPath, 'utf-8'));
+          const [major] = version.split('.');
+          // typescript 5.x
+          if (+major >= 5) enableExperimentalDecorators = true;
+        } catch {
+          this.warn({
+            message: 'Failed to find TypeScript. Please check if TypeScript has been installed.',
+            pluginCode: 'SWC_TYPESCRIPT_NOT_EXISTS'
+          });
+        }
+      }
```

Comment:
Should we run this during every `transform` phase?

IMHO we should hoist this to the plugin initialization.

Reply:
I modified it.